packages/components/src/styles/Button.css.ts

```diff
@@ -0,0 +1,195 @@
+import type { RecipeVariants } from '@vanilla-extract/recipes';
+
+import { vars } from '@launchpad-ui/vars';
+import { recipe } from '@vanilla-extract/recipes';
+
+import { variants as buttonGroup } from './ButtonGroup.css';
+
+const variants = recipe({
+  base: {
+    font: vars.label[1].medium,
+    display: 'inline-flex',
+    alignItems: 'center',
+    justifyContent: 'center',
+    gap: vars.spacing[200],
+    margin: 0,
+    textDecoration: 'none',
+    appearance: 'none',
+    cursor: 'pointer',
+    borderRadius: vars.border.radius.medium,
+    borderStyle: 'solid',
+    borderWidth: '1px',
+    ':focus-visible': {
+      outline: `2px solid ${vars.color.shadow.interactive.focus}`,
+      outlineOffset: '2px',
+    },
+    selectors: {
+      '&[disabled][data-disabled]': {
+        background: 'unset',
+        backgroundColor: vars.color.bg.interactive.disabled,
+        borderColor: vars.color.border.interactive.disabled,
+        color: vars.color.text.interactive.disabled,
+        cursor: 'not-allowed',
+      },
+      [`${buttonGroup.classNames.variants.spacing.compact} &:first-child`]: {
+        borderTopRightRadius: 0,
+        borderBottomRightRadius: 0,
+        borderRight: 'none',
+      },
+      [`${buttonGroup.classNames.variants.spacing.compact} &:last-child`]: {
+        borderTopLeftRadius: 0,
+        borderBottomLeftRadius: 0,
+        borderLeft: 'none',
+      },
+      [`${buttonGroup.classNames.variants.spacing.compact} &:not(:first-child):not(:last-child)`]: {
+        borderRadius: 0,
+      },
```

Comment:
Example of [circular selectors](https://vanilla-extract.style/documentation/styling/#circular-selectors)